_posts/2019-06-04-loading-large-image-data-with-dask-array.md

```diff
@@ -67,160 +173,923 @@ pipeline.
-less information and may notate it differently than we have.
+Let's start again with our image data from the top of the post:
 
 ```
-mydata_ch<i>_<j>t_<k>x_<l>y_<m>z.tif
+$ $ ls raw/ | head
+ex6-2_CamA_ch1_CAM1_stack0000_560nm_0000000msec_0001291795msecAbs_000x_000y_000z_0000t.tif
+ex6-2_CamA_ch1_CAM1_stack0001_560nm_0043748msec_0001335543msecAbs_000x_000y_000z_0000t.tif
+ex6-2_CamA_ch1_CAM1_stack0002_560nm_0087497msec_0001379292msecAbs_000x_000y_000z_0000t.tif
+ex6-2_CamA_ch1_CAM1_stack0003_560nm_0131245msec_0001423040msecAbs_000x_000y_000z_0000t.tif
+ex6-2_CamA_ch1_CAM1_stack0004_560nm_0174993msec_0001466788msecAbs_000x_000y_000z_0000t.tif
 ```
 
-To load a single image, a user might write some code like the following:
+### Load a single sample image with Scikit-Image
+
+To load a single image, we use [Scikit-Image](https://scikit-image.org/):
 
 ```python
-from skimage.io import imread
-a = imread("mydata_ch<i>_<j>t_<k>x_<l>y_<m>z.tif")
+>>> import glob
+>>> filenames = glob.glob("raw/*.tif")
+>>> len(filenames)
+597
+
+>>> import skimage.io
+>>> sample = skimage.io.imread(filenames[0])
+>>> sample.shape
+(201, 1024, 768)
 ```
 
-While this works fine for reading a single image into memory, this will quickly
-become prohibitive once we try to load more data than we have memory. So we
-might use `delayed` to replace this with something like this. We can sub in the
-shape and type from a single file that we have inspected. This could be used
-for multiple files if we know them to be uniform and of the same type. This
-gives us a Dask Array that represents the data on disk for us without using up
-our memory to store it. Pulling from the data on demand.
+Each filename corresponds to some 3d chunk of a larger image.  We can look at a
+few 2d slices of this single 3d chunk to get some context.
 
 ```python
-import dask.array as da
-from dask.delayed import delayed
-from skimage.io import imread
-a = da.from_delayed(delayed(imread)("mydata_ch<i>_<j>t_<k>x_<l>y_<m>z.tif"),
-                    <shape>,
-                    <dtype>)
+import matplotlib.pyplot as plt
+plt.figure(figsize=(10, 10))
+skimage.io.imshow(sample[:, :, 0])
+```
+
+<img src="https://raw.githubusercontent.com/mrocklin/raw-host/gh-pages/images/aollsm-sample-1.png"
+     width="60%">
+
+```python
+plt.figure(figsize=(10, 10))
+skimage.io.imshow(sample[:, :, 0])
 ```
 
-If we want to simplify this a bit, we can use a pre-rolled implementation that
-does this work for us like [dask-image's `imread`](
-https://dask-image.readthedocs.io/en/latest/dask_image.imread.html ). This
-combines the simplicity of an API like scikit-image's with the benefits of
-leveraging Dask to manage the larger data.
+<img src="https://raw.githubusercontent.com/mrocklin/raw-host/gh-pages/images/aollsm-sample-2.png"
+     width="60%">
 
 ```python
-from dask_image.imread import imread
-a = imread("mydata_ch<i>_<j>t_<k>x_<l>y_<m>z.tif")
+plt.figure(figsize=(10, 10))
+skimage.io.imshow(sample[:, :, 0])
```

Comment:
It looks like all of these are `sample[:, :, 0]`, but the viewed data looks notably different. Guessing the slices should be varying between these. Are these take along different axes or something else?